### PR TITLE
[BEAM-10961] Enabled strict dependency on SDKS-Testing

### PR DESCRIPTION
**Pull request by sonam-vend**
Since enabling the strict dependencies on all Java modules is a large and critical change to be merged all at once. Therefore, task [[BEAM-10961](https://issues.apache.org/jira/browse/BEAM-10961)] is divided into sub-tasks and strict dependencies are enabled to each module separately which is comparatively easy to review and merge.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Dataflow | Flink | Samza | Spark | Twister2
--- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/) | ---
Java | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Java11/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Twister2/lastCompletedBuild/)
Python | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python38/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/) | ---
XLang | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Direct/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Dataflow/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/) | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website | Whitespace | Typescript
--- | --- | --- | --- | --- | --- | ---
Non-portable | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocker_Cron/lastCompletedBuild/) <br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocs_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Whitespace_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Typescript_Cron/lastCompletedBuild/)
Portable | --- | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | --- | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by sonam-vend** [comment]
R: @ibzib , @aaltay can you please review or add reviewers who can review this.

File: sdks/java/testing/test-utils/build.gradle [hunk 1]
```diff
@@ -30,12 +31,15 @@ dependencies {
   compile project(path: ":sdks:java:core", configuration: "shadow")
   compile library.java.vendored_guava_26_0_jre
   compile library.java.google_cloud_bigquery
-  compile project(":sdks:java:extensions:google-cloud-platform-core")
+  compile "com.google.code.gson:gson:2.8.6"
+  compile library.java.joda_time
+  compile "org.apache.commons:commons-compress:1.20"
```
Comment (ibzib) [inline sdks/java/testing/test-utils/build.gradle:36]:
```suggestion
  compile library.java.commons_compress
```

File: sdks/java/testing/test-utils/build.gradle [hunk 2]
```diff
@@ -30,12 +31,15 @@ dependencies {
   compile project(path: ":sdks:java:core", configuration: "shadow")
   compile library.java.vendored_guava_26_0_jre
   compile library.java.google_cloud_bigquery
-  compile project(":sdks:java:extensions:google-cloud-platform-core")
+  compile "com.google.code.gson:gson:2.8.6"
```
Comment (ibzib) [inline sdks/java/testing/test-utils/build.gradle:34]:
```suggestion
  compile library.java.google_code_gson
```

File: sdks/java/testing/test-utils/build.gradle [hunk 3]
```diff
@@ -30,12 +31,15 @@ dependencies {
...
+  compile library.java.joda_time
+  compile "org.apache.commons:commons-compress:1.20"
+  compile "org.apache.commons:commons-lang3:3.9"
```
Comment (ibzib) [inline sdks/java/testing/test-utils/build.gradle:37]:
```suggestion
  compile library.java.commons_lang3
```

File: sdks/java/testing/test-utils/build.gradle [hunk 4]
```diff
@@ -30,12 +31,15 @@ dependencies {
...
+  compile "org.apache.httpcomponents:httpclient:4.5.10"
```
Comment (ibzib) [inline sdks/java/testing/test-utils/build.gradle:38]: These don't currently have a common definition in BeamModulePlugin, but they should. BEAM-11669

File: sdks/java/testing/nexmark/build.gradle [hunk 5]
```diff
@@ -62,6 +63,10 @@ dependencies {
   compile project(":sdks:java:extensions:sql:zetasql")
   compile project(":sdks:java:io:kafka")
   compile project(":sdks:java:testing:test-utils")
+  compile "com.google.api-client:google-api-client:1.30.10"
+  compile "junit:junit:4.13-beta-3"
+  compile "org.checkerframework:checker-qual:2.0.0"
```
Comment (ibzib) [inline sdks/java/testing/nexmark/build.gradle:68]:
```suggestion
```

File: sdks/java/testing/nexmark/build.gradle [hunk 6]
```diff
@@ -62,6 +63,10 @@ dependencies {
   compile project(":sdks:java:extensions:sql:zetasql")
   compile project(":sdks:java:io:kafka")
   compile project(":sdks:java:testing:test-utils")
+  compile "com.google.api-client:google-api-client:1.30.10"
+  compile "junit:junit:4.13-beta-3"
```
Comment (ibzib) [inline sdks/java/testing/nexmark/build.gradle:67]:
```suggestion
  compile library.java.junit
```

File: sdks/java/testing/nexmark/build.gradle [hunk 7]
```diff
@@ -62,6 +63,10 @@ dependencies {
   compile project(":sdks:java:extensions:sql:zetasql")
   compile project(":sdks:java:io:kafka")
   compile project(":sdks:java:testing:test-utils")
+  compile "com.google.api-client:google-api-client:1.30.10"
```
Comment (ibzib) [inline sdks/java/testing/nexmark/build.gradle:66]:
```suggestion
  compile library.java.google_api_client
```

File: sdks/java/testing/load-tests/build.gradle [hunk 8]
```diff
@@ -61,14 +62,18 @@ configurations {
...
-
   compile project(path: ":sdks:java:core", configuration: "shadow")
-  compile project(path: ":runners:direct-java", configuration: "shadow")
   compile project(":sdks:java:io:synthetic")
   compile project(":sdks:java:testing:test-utils")
   compile project(":sdks:java:io:google-cloud-platform")
   compile project(":sdks:java:io:kafka")
   compile project(":sdks:java:io:kinesis")
+  compile "com.amazonaws:aws-java-sdk-core:1.11.718"
+  compile "com.google.cloud:google-cloud-core:1.93.7"
```
Comment (ibzib) [inline sdks/java/testing/load-tests/build.gradle:72]:
```suggestion
  compile library.java.google_cloud_core
```

File: sdks/java/testing/load-tests/build.gradle [hunk 9]
```diff
@@ -61,14 +62,18 @@ configurations {
...
+  compile library.java.joda_time
+  compile library.java.vendored_guava_26_0_jre
+  compile "org.checkerframework:checker-qual:2.0.0"
```
Comment (ibzib) [inline sdks/java/testing/load-tests/build.gradle:75]:
```suggestion
```

File: sdks/java/testing/load-tests/build.gradle [hunk 10]
```diff
@@ -61,14 +62,18 @@ configurations {
...
   compile library.java.kafka_clients
-
   compile project(path: ":sdks:java:core", configuration: "shadow")
-  compile project(path: ":runners:direct-java", configuration: "shadow")
   compile project(":sdks:java:io:synthetic")
   compile project(":sdks:java:testing:test-utils")
   compile project(":sdks:java:io:google-cloud-platform")
   compile project(":sdks:java:io:kafka")
   compile project(":sdks:java:io:kinesis")
+  compile "com.amazonaws:aws-java-sdk-core:1.11.718"
```
Comment (ibzib) [inline sdks/java/testing/load-tests/build.gradle:71]:
```suggestion
  compile library.java.aws_java_sdk_core
```

**Review by ibzib — CHANGES_REQUESTED** [review]
Please use the common libraries defined in BeamModulePlugin.groovy wherever possible.

**Comment by sonam-vend** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by sonam-vend** [comment]
Run Python PreCommit

**Review by ibzib — APPROVED** [review]
Looks good, thank you!

**Review by ibzib — CHANGES_REQUESTED** [review]
We can/should probably remove these lines now that junit and hamcrest are compile dependencies.

https://github.com/apache/beam/blob/d2e1f693edbd06a54d703e3a27897a0b42319fe8/sdks/java/testing/nexmark/build.gradle#L73-L74

**Comment by sonam-vend** [comment]
Run Python PreCommit

**Comment by sonam-vend** [comment]
Run Java PreCommit

**Comment by ibzib** [comment]
Run XVR_Flink PostCommit

**Comment by ibzib** [comment]
Run XVR_Spark PostCommit

**Pull request merged**